config/lobby/db/003_add_bcrypt_password_column

```diff
@@ -0,0 +1,17 @@
+start transaction;
+
+alter table ta_users add column bcrypt_password character(60) check (char_length(bcrypt_password)=60);
+alter table ta_users alter column password drop not null;
+update ta_users set password=null, bcrypt_password=password where password like '$2a$%';
+alter table ta_users alter column password type character(34);
+alter table ta_users add constraint ta_users_password_check check (char_length(password)=34);
+alter table ta_users add constraint ta_users_check check (password IS NOT NULL OR bcrypt_password IS NOT NULL);
+
+-- Comments
+comment on column ta_users.password is 'The legacy MD5Crypt hash of the password. Is going to replaced by the content bcrypt_password in the next incompatible lobby release. The length of the hash must always be 34 chars. Either password or bcrypt_password must be not null.';
+comment on column ta_users.bcrypt_password is 'The BCrypt-Hashed password of the user, should be the same as the md5 password but in another form. The length of the hash must always be 60 chars. Either password or bcrypt_password must be not null.';
+comment on constraint ta_users_password_check on ta_users is 'This check constraint ensures the legacy password hash has the right char length of 34 chars';
+comment on constraint ta_users_bcrypt_password_check on ta_users is 'This check constraint ensures the legacy password hash has the right char length of 60 chars';
```

Comment:
Is there some Postgres magic going on when you add the constraint as part of the `ADD COLUMN` above?  I don't see where this constraint name is ever defined.

Reply:
Postgres does actually use a pretty consistent naming scheme AFAIK:
```
CREATE TABLE tableName (
    column1 varchar(1) check (1=1),
    column2 varchar(1),
    check(column1=column2)
);
```
creates 2 check sonstraints:
`tableName_column1_check` which checks 1=1 and `tableName_check` which checks column1=column2
I must admit that I don't know how this behaves with multiple "general, non column specific" checks per table, but this is how it works for those cases.
That's why I didn't change the name, I just wanted to be consistent.

IMO a describing name is not that important, we shouldn't fiddle around with the DB on our own anyways, and the engine shouldn't even trigger those errors, but check the inserted stuff before actually trying to insert the values.
It's just important to know which exact constraint caused the error so after looking up what it does it can be fixed.